File: my-solutions/0242-valid-anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        dict_s = {}
        for i in range(len(s)):
            if s[i] in dict_s:
                dict_s[s[i]] += 1
            else:
                dict_s[s[i]] = 1

        for i in range(len(t)):
            if t[i] not in dict_s:
                return False
            else:
                dict_s[t[i]] -= 1
                if dict_s[t[i]] < 0:
                    return False

        for key in dict_s:
            if dict_s[key] != 0:
                return False

        return True

File: my-solutions/0242-valid-anagram/solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> char_count;
        for(int i = 0; i < size(s); i++){
            if(char_count.find(s[i]) == char_count.end()){
                char_count.insert(pair<char, int>(s[i], 1));
            } else {
                char_count[s[i]]++;
            }
        }
        for(int i = 0; i < size(t); i++) {
            if(char_count.find(t[i]) == char_count.end()) {
                return false;
            } else {
                char_count[t[i]]--;
                if(char_count[t[i]] < 0) {
                    return false;
                }
            }
        }
        for(auto &i: char_count) {
            if(i.second != 0) {
                return false;
            }
        }
        return true;
    }
};
